sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCWriteSuite.scala

```diff
@@ -44,9 +44,6 @@ class JDBCWriteSuite extends SharedSQLContext with BeforeAndAfter {
 
   val testH2Dialect = new JdbcDialect {
     override def canHandle(url: String) : Boolean = url.startsWith("jdbc:h2")
-    override def getCatalystType(
```

Comment:
If we get rid of this, how can it pass the test case ["Remap types via JdbcDialects"](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala#L619). At least, it failed in my local environement. 

This overwrite was intentionally introduced for testing JdbcDialect. See the PR: https://github.com/apache/spark/pull/5498